source/javascripts/components/YmlStorageSettings/YmlStorageSettingsNotifications.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { FC } from "react";
+import { Text, Notification } from "@bitrise/bitkit";
+import InlineLink from "../common/InlineLink";
+import { MonolithError } from "../../hooks/api/useMonolithApiCallback";
+
+export const CreatingYmlOnWebsiteProgress: FC = () => (
+	<Notification type="progress">Creating bitrise.yml on bitrise.io...</Notification>
+);
+
+export const LookingForYmlInRepoProgress: FC = () => (
+	<Notification type="progress">Looking for bitrise.yml in the app repository...</Notification>
+);
+
+export const ValidatingYmlInRepoProgress: FC = () => (
+	<Notification type="progress">Validating bitrise.yml in the app repository...</Notification>
+);
+
+export const YmlNotFoundInRepositoryError: FC = () => (
+	<Notification type="alert">
+		Couldn't find a bitrise.yml file in the app's repository. Add the file to your main branch and try again.
+	</Notification>
+);
+
+export const YmlInRepositoryInvalidError: FC = () => (
+	<Notification type="alert">
+		<Text>
+			The bitrise.yml file in the repository seems invalid. Please review and fix it before proceeding. Read more about
+			the{" "}
+			<InlineLink
+				underline
+				color="red-4"
+				text="valid syntax of the bitrise.yml file."
+				url="https://devcenter.bitrise.io/builds/bitrise-yml-online/"
+			/>
+		</Text>
+	</Notification>
+);
```

Comment:
Most of these components just wrap an already simple component, not much functionality is encapsulated. Is it worth it instead of just using the encapsulated content in the parent component instead? It might be a valid design pattern though, not sure.

Reply:
I was deliberating the same thing when i did these. My "pro" arguments were:

- we only have to maintain the look/text of these messages in one place
- make the components that use these a bit smaller (they have quite some logic in there already, sparing couple of lines makes it cleaner for me)

The first one to move here was `YmlInRepositoryInvalidError` as it has specific styling tied to it, then I decided to move the other "common" notifications as well.

I don't really have a strong preference here, we can remove these changes if you feel like it's unnecessary

Reply:
I can see both solutions being reasonable, we can keep it. I think this should be a part of a bigger discussion when we're planning the ideal structure & arrangement of the project later.